internal/zinc-persist/src/main/java/xsbti/compile/analysis/ReadMapper.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Zinc - The incremental compiler for Scala.
+ * Copyright 2011 - 2017, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * This software is released under the terms written in LICENSE.
+ */
+
+package xsbti.compile.analysis;
+
+import sbt.internal.inc.mappers.RelativeReadMapper;
+import xsbti.compile.MiniSetup;
+
+import java.io.File;
+import java.nio.file.Path;
+
+/**
+ * Defines a reader-only mapper interface that is used by Zinc after reading
+ * the analysis from an instance of an {@link xsbti.compile.AnalysisStore}.
+ *
+ * This interface is useful to make the analysis file machine-independent and
+ * allow third parties to distribute them around.
+ */
+public interface ReadMapper extends GenericMapper {
+
+    /**
+     * Defines a mapper that reads from a machine-independent analysis file.
+     *
+     * An analysis file is machine independent if all the paths are relative to a path
+     * and no information about the machine that produced it is stored -- only
+     * information about the structure of the project from a given build tool is persisted.
+     *
+     * The mapper makes sure that the analysis file looks like if it was generated by
+     * the local machine it's executed on.
+     *
+     * <b>Important note</b>: The assumption that all paths can be made relative to a concrete
```

Comment:
I addressed your concerns here: https://github.com/sbt/zinc/issues/320. Can you have a look?